### PR TITLE
Parse and save data

### PR DESCRIPTION
**Pull request by Flix6x**

### CODE REVIEW

**Comment by Flix6x** [comment]
I just noticed I forgot to add the logic to disregard beliefs that didn't actually change (these just clutter our database). We're getting a lot of those.

**Comment by nhoening** [comment]
Would that logic be best placed more centrally, like FlexMeasures or timely-beliefs?

**Comment by Flix6x** [comment]
See https://github.com/SeitaBV/flexmeasures/pull/183 for the util function.

File: __init__.py [hunk 1]
```diff
@@ -31,22 +42,156 @@
...
+                current_app.logger.error(
+                    f"No sensor set up for Z-info sensor name {zinfo_sensor_name} ..."
+                )
+                continue
+            sensor = sensor_dict[sensor_name]
+            df_sensor = df.query(
+                f"{zinfo_sensor_name_field} == '{zinfo_sensor_name}'"
+            ).droplevel(zinfo_sensor_name_field)[zinfo_event_value_field]
+
+            # required by timely_beliefs, TODO: check if that still is the case, see https://github.com/SeitaBV/timely-beliefs/issues/64
```
Comment (nhoening) [inline __init__.py:132]: That issue is about initializing a BDF with a Series. Here, it seems you initialize with a DataFrame, so maybe this is not needed.

File: __init__.py [hunk 2]
```diff
@@ -31,22 +42,156 @@
...
+            if zinfo_sensor_name not in zinfo_sensor_mapping:
+                continue
+            sensor_name = zinfo_sensor_mapping[zinfo_sensor_name]["sensor_name"]
+            if sensor_name not in sensor_dict:
+                current_app.logger.error(
+                    f"No sensor set up for Z-info sensor name {zinfo_sensor_name} ..."
+                )
+                continue
+            sensor = sensor_dict[sensor_name]
+            df_sensor = df.query(
```
Comment (nhoening) [inline __init__.py]: Is this duplicated effort? You already built a list of `df_sensor` objects in the previous loop.

**Review by nhoening — APPROVED** [review]
Nice work, only small comments.

I guess it would be a good time to add some helicopter description how Z-Info works to the Readme.
What I find conceptually noteworthy is that we are configuring sensors we expect, and then have to check if those are the sensors we receive back. This is because the Z-Info host has their own configuration which sensors we are allowed to read, and we always (at least in the current state) get all of these back. 
Am I seeing this correctly?

File: __init__.py (hunk 1 above)
Comment (Flix6x) [reply __init__.py:132]: It was needed for initializing with a DataFrame, too.

**Pull request merged**